cmd/duffle/main.go

```diff
@@ -130,17 +131,38 @@ type driverWithRelocationMapping struct {
 }
 
 func (d *driverWithRelocationMapping) Run(op *driver.Operation) error {
-	// if there is a relocation mapping, ensure it is mounted
+	// if there is a relocation mapping, ensure it is mounted and relocate the invocation image
 	if d.relMapping != "" {
 		op.Files["/cnab/app/relocation-mapping.json"] = d.relMapping
+
+		if err := d.relocateImage(&op.Image); err != nil {
+			return err
+		}
 	}
+
 	return d.driver.Run(op)
 }
 
 func (d *driverWithRelocationMapping) Handles(it string) bool {
 	return d.driver.Handles(it)
 }
 
+func (d *driverWithRelocationMapping) relocateImage(im *string) error {
+	relMap := make(map[string]string)
+	err := json.Unmarshal([]byte(d.relMapping), &relMap)
+	if err != nil {
+		return err
```

Comment:
I suggest wrapping this error to provide some context on where the error came from. Otherwise if it fails on a json unmarshal, the user may get a pretty unhelpful error mesage and it won't be clear that it is due to the relocation mapping.

Reply:
Good idea. Fixed.